mmv1/third_party/terraform/tpgresource/utils.go

```diff
@@ -702,3 +741,48 @@ func GetContentMd5Hash(content []byte) string {
 	}
 	return base64.StdEncoding.EncodeToString(h.Sum(nil))
 }
+
+func DefaultProviderProject(_ context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+
+	config := meta.(*transport_tpg.Config)
+
+	//project
+	if project := diff.Get("project"); project != nil {
+		project, err := GetProjectFromDiff(diff, config)
+		if err != nil {
+			return fmt.Errorf("Failed to retrieve project, pid: %s, err: %s", project, err)
+		}
+		diff.SetNew("project", project)
```

Comment:
If this call returns an error we should capture + return it! That will catch non-Computed fields.